File: newsmodule/src/main/java/zjm/cst/dhu/newsmodule/utils/ItemTouchHelperCallBack.java
package zjm.cst.dhu.newsmodule.utils;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;

import java.util.Collections;
import java.util.List;

import zjm.cst.dhu.library.event.ChannelChangeEvent;
import zjm.cst.dhu.library.mvp.model.NewsChannel;
import zjm.cst.dhu.library.utils.RxBus;
import zjm.cst.dhu.newsmodule.mvp.ui.adapter.ChannelManageLikeAdapter;

/**
 * Created by zjm on 2017/5/12.
 */

public class ItemTouchHelperCallBack extends ItemTouchHelper.Callback {


    public ItemTouchHelperCallBack() {
    }

    @Override
    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
        return makeMovementFlags(ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT | ItemTouchHelper.UP | ItemTouchHelper.DOWN
                | ItemTouchHelper.START | ItemTouchHelper.DOWN, 0);
    }

    @Override
    public boolean isItemViewSwipeEnabled() {
        return false;
    }

    @Override
    public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
        int fromPosition = viewHolder.getAdapterPosition();//得到拖动ViewHolder的position
        int toPosition = target.getAdapterPosition();//得到目标ViewHolder的position
        RxBus.getDefault().post(new ChannelChangeEvent(true, fromPosition, toPosition));
        ChannelManageLikeAdapter.HolderChannelLike holderChannelLike =
                (ChannelManageLikeAdapter.HolderChannelLike) viewHolder;
        holderChannelLike.moduleNewsItemChannelLikeBinding.setIsManage(false);
        return true;
    }

    @Override
    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {

    }
}

File: newsmodule/src/main/kotlin/zjm.cst.dhu.newsmodule/utils/ItemTouchHelperCallBack.kt
package zjm.cst.dhu.newsmodule.utils

import android.support.v7.widget.RecyclerView
import android.support.v7.widget.helper.ItemTouchHelper

import zjm.cst.dhu.library.event.ChannelChangeEvent
import zjm.cst.dhu.library.utils.others.RxBus
import zjm.cst.dhu.newsmodule.mvp.ui.adapter.ChannelManageLikeAdapter


/**
 * Created by zjm on 2017/5/12.
 */

class ItemTouchHelperCallBack
    : ItemTouchHelper.Callback() {

    override fun getMovementFlags(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder):
            Int = ItemTouchHelper.Callback.makeMovementFlags(ItemTouchHelper.LEFT
            or ItemTouchHelper.RIGHT or ItemTouchHelper.UP
            or ItemTouchHelper.DOWN or ItemTouchHelper.START or ItemTouchHelper.DOWN, 0)


    override fun isItemViewSwipeEnabled():
            Boolean = false


    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder,
                        target: RecyclerView.ViewHolder):
            Boolean {
        val fromPosition = viewHolder.adapterPosition//得到拖动ViewHolder的position
        val toPosition = target.adapterPosition//得到目标ViewHolder的position
        RxBus.default.post(ChannelChangeEvent(isMove = true, fromPosition = fromPosition,
                toPosition = toPosition))
        val holderChannelLike = viewHolder as ChannelManageLikeAdapter.HolderChannelLike
        holderChannelLike.moduleNewsItemChannelLikeBinding.isManage = false
        return true
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {

    }
}
